Core/Src/currentSense.c

```diff
@@ -0,0 +1,59 @@
+/* ==================================================================== */
+/* ============================= INCLUDES ============================= */
+/* ==================================================================== */
+
+#include "currentSense.h"
+#include "GopherCAN.h"
+#include <math.h>
+#include "debug.h"
+
+/* ==================================================================== */
+/* =================== GLOBAL FUNCTION DEFINITIONS ==================== */
+/* ==================================================================== */
+
+/*!
+  @brief   Update the bms tractive system current data and sensor status from gophercan
+  @param   bms - BMS data struct
+*/
+void getTractiveSystemCurrent(Bms_S* bms)
+{
+    // Fetch current sensor data from gophercan
+    float currHI = bmsTractiveSystemCurrentHigh_A.data;
+    float currLO = bmsTractiveSystemCurrentLow_A.data;
+
+    // If the current exceeds the following threshold in either the positive or negative direction,
+    // the sensor input has railed to 0 or 5v and a current sensor error is set
+    if(fabs(currHI) < CURRENT_HIGH_RAIL_THRESHOLD)
+    {
+        bms->currentSensorStatusHI = SENSE_GOOD;
+    }
+    else
+    {
+        // Debug("Current sensor high channel has railed!\n");
+        bms->currentSensorStatusHI = SENSE_MIA;
+    }
+    
+    if(fabs(currLO) < CURRENT_LOW_RAIL_THRESHOLD)
+    {
+        bms->currentSensorStatusLO = SENSE_GOOD;
+    }
+    else
+    {
+        // Debug("Current sensor low channel has railed!\n");
+        bms->currentSensorStatusLO = SENSE_MIA;
+    }
+
+    // To use the HI current sensor channel, it must be working AND (it must exceed the measuring range of the low channel OR the low channel must be faulty)
+    if ((bms->currentSensorStatusHI == SENSE_GOOD) && ((fabs(currLO) > CURRENT_LOW_TO_HIGH_SWITCH_THRESHOLD) || (bms->currentSensorStatusLO != SENSE_GOOD)))
```

Comment:
We might want to do some sort of linear interpolation between the two values when we are switching over, just so that we don't get a sharp jump when we are at the threshold. Also if the current sensor is hovering around the 75A mark we could get a lot of noise as we switch between the low and the high channel. Maybe between 73A-77A we should make the calculated current as follows: At 73A use entirely the low channel and at 77A entirely use the high channel. At 75A we use 50% of the low and 50% of the high channel. Lmk what you think